### DISCUSSION TITLE
rtsp stream freeze

### DISCUSSION THREAD

**Discussion by avinoam31** [bluenviron/mediamtx#935] [Questions]
hello,
i am using rtsp-smple-server to stream rtsp from live camera and recently start having the stream freeze for a bit every once in a short while.
my system is as follows:
camera -> server -> pc
the server receive the live video from the camera, do its magic and send the video by ffmpeg to simple, which sits on the same machine (windows). there is another route for the video (not via simple/ffmpeg) which don't have any freezing so those aren't from the camera.
i connected to the server via remote control, opened up a vlc on both the pc and the server and compared them.
both get a freeze here and there but not at the same time - so that is not a network problem (simple lan), and also not a problem of the ffmpeg transmission or something before it, as the second vlc shows the frames that the freezed one got stuck on.

i tried upgrading simple from 0.17.9 to 0.18.2 with no luck.

my simple yml only changed the transmit protocol to allow only TCP. also tried to increase readBufferCount.
ffmpeg cmd:
ffmpeg -pix_fmt ___ -video_size __x__ -an -f raw_video -i pipe:0 -pix_fmt yuv420p -preset ultrafast -vcodec libx264 -tune zerolatency -stmeout 50000 -f rtsp rtps://127.0.0.1:8554/streamName

will be glad for any help.
tnx

**Comment by avinoam31** [comment]
update - i think the problem is actually with vlc. i tried opening ffplay and it didnt have those freezes.  

does anyone knows a good free player for watching rtsp? smallest latency possible preferred as it should show live stream

**Reply by aler9** [reply]
A good cross-compatible player is MPV:
https://mpv.io/

**Reply by osenbal** [reply]
theres any solution for libvlc parameter ?

**Reply by SixK** [reply]
maybe you should try to reduce vlc buffering or buffer more longer (but this mean more latency) ? By default vlc buffer 2 seconds of video... You may also try to add more I frame. It could be also a problem between udp/tcp preference ? But I think ffmpeg and vlc use udp by default.
